Removes change note

elenatanasoiu opened the pull request:
This will remove a change note added in error on a publication:

https://www.gov.uk/government/publications/sen-and-disability-reform-s31-grant-determinations#history

The change was requested in this zendesk ticket: https://govuk.zendesk.com/agent/tickets/2566949

For: https://trello.com/c/sOPRBrrT/63-2-remove-change-note-and-redirect-html-attachment-url

thomasleese commented on db/migrate/20180215112954_fix_whitehall_remove_change_note.rb:
```diff
@@ -0,0 +1,15 @@
+    change_notes = d.editions.map(&:change_note).compact
+    change_notes.select{ |change_note| change_note.note == "Added funding allocations for individual LAs for 2018 to 2019."}
+    # Re-present editions to content store
+    puts "The editions that need to be represented downstream are: #{change_notes.map(&:edition_id)}"
```
It is actually actually possible to represent stuff from the migration, see https://github.com/alphagov/publishing-api/blob/fab6a131d52b9759160b4004f9e11953dad50ba7/db/migrate/20171011154900_fix_remaining_specialist_publisher_first_published_at_dates.rb#L13 for some prior art here. An important thing to note is the `disable_ddl_transaction!` at the top.

elenatanasoiu replied:
Thanks!

elenatanasoiu commented:
I've added the `schema.rb`.

I've also added a link to the Zendesk ticket requiring the change. In a nutshell an HTML attachment was added to a different document by mistake and we've had to make a redirect for it. The person who added the original attachment also added a change note which is visible in the UI and is no longer accurate. Initially we were provided with the whitehall-admin link of the document and asked to remove the latest change note. This was actually not the same note as the latest one visible in the UI, which is the note they were actually referring to, but hadn't specified. 

So now we're removing the visible UI change note as requested.

thomasleese commented on db/migrate/20180215112954_fix_whitehall_remove_change_note.rb:
```diff
@@ -0,0 +1,27 @@
+      change_notes.select{ |change_note| change_note.note == "Added funding allocations for individual LAs for 2018 to 2019."}
+  
+      # Get rid of change notes
+      change_notes.map(&:destroy)
```
I wonder if this might delete all the change notes because the `select` call above doesn't get reassigned to the `change_notes` variable?

elenatanasoiu replied:
Thanks! Corrected here: https://github.com/alphagov/publishing-api/pull/1166
